### ISSUE TITLE
WhatsApp always disconnects after a few minutes and never connects again

### ISSUE THREAD

**Issue by Sulivan1999** [adiwajshing/Baileys#1233]
Hi, I have a big question. I was using a number normally connected to Baileys until one day it stopped connecting and didn't generate any errors in the terminal. So I tested it directly on WhatsApp Web through the browser and it also doesn't keep in touch for more than 2 minutes, the connection drops all the time.

I did everything I could from checking the phone's battery saving mode, checking the app running in the background, testing on different internet networks, clearing the app cache, deleting the WhatsApp account and starting all over again. . But it didn't help, so I bought another number and activated a new WhatsApp account on another phone which took a few more hours and I started having the same problem.

Note: there have already been 4 numbers that after being connected to Baileys had this same problem, I don't know what to do. I tested a new number on WhatsApp only and there was no problem. The number is not banned, it just stops connecting to WhatsApp Web and it takes time to send and receive messages on the cell phone itself.

Sorry for my english.

**Comment by AkirahX** [comment]
One question, are you using an iPhone or any other Apple device?

**Comment by Sulivan1999** [comment]
No, my devices are Android. One detail, I don't use any unofficial APK or WhatsApp app clone and I have tested the same issue on both Personal and Business WhatsApp as well.

**Comment by AkirahX** [comment]
> No, my devices are Android. One detail, I don't use any unofficial APK or WhatsApp app clone and I have tested the same issue on both Personal and Business WhatsApp as well.

Could you provide the complete connection code?  You may be configuring something wrong.

**Comment by canove** [comment]
Maybe I had this same problem a few times.

When It occurs, the whatsapp stop working well even on Phone, even without baileys or whatsapp web connected.

In my cases, it seems like it stops receiving some events from whatsapp server. 

I'm able to send messages through phone, but I never get the ACK back and can't receive messages as well. 

When someone tries to send a message to this bugged number, they can see my online status, but the sent message gets only the first ack ✔️

It normally get back working after a few hours. 

Does it looks like to be the same problem?

**Comment by Sulivan1999** [comment]
> 

Actually I'm using an API that was built on top of Baileys. I didn't make any changes other than the name of the browser that appears on the phone when it is connected.

This is in the API repository: https://github.com/salman0ansari/whatsapp-api-nodejs

**Comment by Sulivan1999** [comment]
> Talvez eu tive esse mesmo problema algumas vezes.
> 
> Quando isso ocorre, o whatsapp deixa de funcionar bem mesmo no telefone, mesmo sem baileys ou whatsapp web conectado.
> 
> Nos meus casos, parece que ele deixa de receber alguns eventos do servidor do whatsapp.
> 
> Consigo enviar mensagens pelo telefone, mas nunca recebo o ACK e não consigo receber mensagens também.
> 
> Quando alguém tenta enviar uma mensagem para este número grampeado, ele pode ver meu status online, mas a mensagem enviada recebe apenas a primeira confirmação✔️
> 
> Normalmente volta a funcionar depois de algumas horas.
> 
> Parece ser o mesmo problema?

Yea! Exactly the same problem, it happens with every number I plug in and it doesn't work again after a few hours or days.

**Comment by frontln** [comment]
I too face the same problem.
For me baileys api was working fine until a few days ago.
And suddenly I was told by my friends that I am not receiving messages on whatsapp.
This is what is happening.
Bailey  shows qr and upon scanning
The device connects then immediately loses connection. Linked devices shows Last active a minute ago.
Strangely after this WhatsApp on the connected phone also starts doing vague things like if someone sends the affected phone a message it shows only single tick on the other phone and no messages are received.
The messages sent from the connected phone also is not received by the other end.
Are there any breaking changes?

**Comment by ReiDx** [comment]
same problem

**Comment by phenriquec** [comment]
same problem

**Comment by canove** [comment]
I don't think its related to some changes in Baileys, cause I'm using the old version (3.5.3) and first time I saw this behave was months ago. 

On affected phone, switching to Multidevice (Beta) makes phone start to receive message again, but leaving beta make it stop right away. 

Every time it happens, the only solution I found is to wait a few hours until phone start receiving messages again.

**Comment by Sulivan1999** [comment]
After a lot of testing I noticed some points that lead to this problem (at least for me).

I first noticed that if you made a connection from multiple Bailey sessions on the same WhatsApp account, it gets to a point where each session keeps trying to connect and one session replaces the other, causing WhatsApp to have the described issue at the beginning of this discussion.

I also noticed that if I have two sessions or more connected at the same time, the problem may not occur immediately, but if I try to send a message through one of the sessions (with the cell phone connected or not to the internet) it can happen that the problem exists.

Finally, before testing different WhatsApp numbers when the problem happened, for me none of the numbers worked again after a few hours, days or weeks. Now I found that if I delete the WhatsApp account completely in "Settings > Account > Delete my account" I can create a new one on the same number and then it comes back without the problem, that was my only solution besides taking care to check the sessions before activate a new one or before sending a message.
